### ISSUE TITLE
seems not working with docker 1.7

### ISSUE THREAD

**Issue by suppix** [Kloadut/dokku-pg-plugin#77]
dokku postgresql:create db
Your docker version is too old, please update it

dokku postgresql:link web db
docker: "inspect" requires a minimum of 1 argument.
See 'docker inspect --help'.

Usage: docker inspect [OPTIONS] CONTAINER|IMAGE [CONTAINER|IMAGE...]

**Comment by Subtletree** [comment]
+1

**Comment by dianseptiana** [comment]
I have the same problem. It keep saying my docker version is too old, when i run "dokku postgresql:create appdb". This is the detail of installed docker:

Client version: 1.7.0
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 0baf609
OS/Arch (client): linux/amd64
Server version: 1.7.0
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 0baf609
OS/Arch (server): linux/amd64

**Comment by masodev** [comment]
Same issue here!

**Comment by stephenbaidu** [comment]
Same here. I have Docker 1.7.0

**Comment by stephenbaidu** [comment]
According to the Docker release note (Jun 22), the volumes systems have been completely rewritten. After a shallow search, I realised `/var/lib/docker/vfs/dir/` is no more. So I changed
`VOLUME_PATH=$(docker inspect $ID | grep /var/lib/docker/vfs/dir/ | awk '{print $2}' | sed -e's/"//g')`
to
`VOLUME_PATH=$(docker inspect $ID | grep /var/lib/docker/volumes | awk '{print $2}' | sed -e's/"//g')`

And I get the following error

```
dokku postgresql:create myapp
docker: "inspect" requires a minimum of 1 argument.
See 'docker inspect --help'.

Usage: docker inspect [OPTIONS] CONTAINER|IMAGE [CONTAINER|IMAGE...]

Return low-level information on a container or image
json: cannot unmarshal array into Go value of type struct { NetworkSettings struct { Ports nat.PortMap } }
-----> Setting config vars
       DATABASE_URL: postgres://root:dCy8p6GLF5gCUl7K@:/db
-----> Restarting app myapp
App myapp has not been deployed

-----> myapp linked to postgresql/myapp database

-----> PostgreSQL container created: postgresql/myapp
docker: "inspect" requires a minimum of 1 argument.
See 'docker inspect --help'.

Usage: docker inspect [OPTIONS] CONTAINER|IMAGE [CONTAINER|IMAGE...]

Return low-level information on a container or image
json: cannot unmarshal array into Go value of type struct { NetworkSettings struct { Ports nat.PortMap } }

       Host: 
       Port: 
       User: 'root'
       Password: 'dCy8p6GLF5gCUl7K'
       Database: 'db'

       Url: 'postgres://root:dCy8p6GLF5gCUl7K@:/db'
```

**Comment by kriskova** [comment]
+1

**Comment by stephenbaidu** [comment]
I was able to fix this at my end. I made two changes in the [commands](https://github.com/stephenbaidu/dokku-pg-plugin/blob/master/commands) file.
There has been a change in the volumes system, so I changed

```
VOLUME_PATH=$(docker inspect $ID | grep /var/lib/docker/vfs/dir/ | awk '{print $2}' | sed -e's/"//g')
```

to

```
VOLUME_PATH=$(docker inspect $ID | grep /var/lib/docker/volumes | awk '{print $2}' | sed -e's/"//g')
```

And since the created postgresql/$APP image does not have the `latest` tag added, I changed

```
function get_postgresql_id() {
    ID=$(docker ps | grep postgresql/"$APP":latest | awk {'print $1'})
```

to

```
function get_postgresql_id() {
    ID=$(docker ps | grep postgresql/"$APP" | awk {'print $1'})
```

This works for me for now.

**Comment by kriskova** [comment]
@stephenbaidu +1! Works for me! Thanks!

**Comment by srigi** [comment]
+1 for integrating @stephenbaidu's patch

**Comment by egerlach** [comment]
You can also pull from [my branch](/egerlach/dokku-pg-plugin/issue-77) until it is merged. It incorporates @stephenbaidu's changes.

**Comment by slava-vishnyakov** [comment]
:+1: for the fix from @stephenbaidu , thanks!

**Comment by srigi** [comment]
It looks like this project is abandoned by original author. Can you suggest good alternative?

**Comment by dmr** [comment]
Cool, @srigi did you find any?

**Comment by masodev** [comment]
My issues were resolved after upgrading. Please refer to:
https://github.com/Kloadut/dokku-pg-plugin/issues/76#issuecomment-124716847

**Comment by srigi** [comment]
@dmr yes, I'm really happy with [Flink/dokku-psql-single-container](https://github.com/Flink/dokku-psql-single-container). Don't be scared that there isn't any `link` command. It will auto-link to app container with the same name.
